actionpack/lib/action_dispatch/journey/gtg/transition_table.rb

```diff
@@ -164,8 +164,12 @@ def []=(from, to, sym)
           to_mappings = states_hash_for(sym)[from] ||= {}
           case sym
           when Regexp
-            # we must match the whole string to a token boundary
-            sym = Regexp.new(/\A#{sym}\Z/)
+            if sym == DEFAULT_EXP
+              sym = DEFAULT_EXP_ANCHORED
+            else
+              # we must match the whole string to a token boundary
```

Comment:
This comment seems to apply to the `DEFAULT_EXP` case too, it should probably stay where it was.

Reply:
Sure.